src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -559,28 +591,99 @@ Namespace Microsoft.CodeAnalysis.Semantics
+                boundConversion.Type.IsDelegateType()) Then
+                ' If this is a conversion from a lambda to a delegate type, or this is an AddressOf delegate creation
+                ' as determined above, we return a DelegateCreationExpression, instead of returning a conversion expression
+                Return New LazyDelegateCreationExpression(operand, _semanticModel, syntax, type, constantValue, isImplicit)
             End If
 
-            Dim conversion = New Conversion(New KeyValuePair(Of ConversionKind, MethodSymbol)(boundConversion.ConversionKind, methodSymbol))
+            Dim conversion = New Conversion(New KeyValuePair(Of VisualBasic.ConversionKind, MethodSymbol)(boundConversion.ConversionKind, methodSymbol))
             Dim isExplicit As Boolean = boundConversion.ExplicitCastInCode
             Dim isTryCast As Boolean = False
             Dim isChecked As Boolean = False
-            Dim type As ITypeSymbol = boundConversion.Type
-            Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundConversion.ConstantValueOpt)
-            Dim isImplicit As Boolean = boundConversion.WasCompilerGenerated
             Return New LazyVisualBasicConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit)
         End Function
 
+        Private Function IsValidDelegateCreationSyntax(conversionKind As ConversionKind, conversionSyntax As SyntaxNode, operandSyntax As SyntaxNode) As Boolean
+            ' An identity conversion is introduced by the compiler on top of New DelegateType(AddressOf Method). This node
+            ' only exists for the convenience of SemanticModel, so we want to classify it as a Delegate Creation
+            Dim validConversionKind = conversionKind = ConversionKind.Identity OrElse
+                                      (conversionKind And (Not ConversionKind.DelegateRelaxationLevelMask)) = 0
+
+            ' Any of the explicit cast types, as well as New DelegateType(AddressOf Method)
+            Dim validConversionSyntax = conversionSyntax.Kind() = SyntaxKind.CTypeExpression OrElse
+                                        conversionSyntax.Kind() = SyntaxKind.DirectCastExpression OrElse
+                                        conversionSyntax.Kind() = SyntaxKind.TryCastExpression OrElse
+                                        conversionSyntax.Kind() = SyntaxKind.ObjectCollectionInitializer
+
+            Dim validOperandSyntax = operandSyntax.Kind() = SyntaxKind.AddressOfExpression
+
+            Return validConversionKind AndAlso validConversionSyntax AndAlso validOperandSyntax
+        End Function
+
+        Private Function CreateConversionOperation(operand As BoundNode, conversionKind As ConversionKind, conversionSyntax As SyntaxNode) As (methodSymbol As MethodSymbol, operation As Lazy(Of IOperation), isAddressOfDelegateCreation As Boolean)
```

Comment:
>Private Function CreateConversionOperation(operand As BoundNode, conversionKind As ConversionKind, conversionSyntax As SyntaxNode) As (methodSymbol As MethodSymbol, operation As Lazy(Of IOperation), isAddressOfDelegateCreation As Boolean) [](start = 8, length = 238)

This line is too long. It is also hard to read because it elements of the returned tuple look like parameters of the method. Consider putting the return type on a separate line. Something like:
```
Private Function CreateConversionOperation( ...
) As (methodSymbol As MethodSymbol, operation As Lazy(Of IOperation), isAddressOfDelegateCreation As Boolean)
...
```

Reply:
Done.

---
In reply to: [142262835](https://github.com/dotnet/roslyn/pull/22056#discussion_r142262835) [](ancestors = 142262835)